From ac07d1deb34191b71dcf2f6615fba14a0eb55aad Mon Sep 17 00:00:00 2001
From: Aleksander Jan Bajkowski <olek2@wp.pl>
Date: Sun, 28 Sep 2025 23:31:00 +0200
Subject: [PATCH] test

---
 src/mdio/mdio.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/src/mdio/mdio.c b/src/mdio/mdio.c
index 4677db6..8ceaed5 100644
--- a/src/mdio/mdio.c
+++ b/src/mdio/mdio.c
@@ -80,15 +80,23 @@ static int msg_query(struct nlmsghdr *nlh, mnl_cb_t callback, void *data)
 	struct mnl_socket *nl;
 	int ret;
 
+	printf("%s:1\n", __func__);
+
 	seq = time(NULL);
 	nlh->nlmsg_seq = seq;
 
+	printf("%s:2\n", __func__);
+
 	nl = msg_send(NETLINK_GENERIC, nlh);
 	if (!nl)
 		return -ENOTSUP;
 
+	printf("%s:3\n", __func__);
+
 	ret = msg_recv(nl, callback, data, seq);
 	mnl_socket_close(nl);
+
+	printf("%s:4 ret=%d\n", __func__, ret);
 	return ret;
 }
 
@@ -519,10 +527,12 @@ int mdio_common_dump_exec_one(struct mdio_device *dev, int *argc, char ***argv)
 	uint32_t reg;
 	int err;
 
+	printf("%s:1\n", __func__);
 	err = mdio_device_parse_reg(dev, argc, argv, &range.start, &range.end);
 	if (err)
 		return err;
 
+	printf("%s:2\n", __func__);
 	/* Can't emit a loop, since there's no way to pass the (mdio)
 	 * register in a (mdio-netlink) register - so we unroll it. */
 	for (reg = range.start; reg <= range.end; reg++) {
@@ -533,6 +543,8 @@ int mdio_common_dump_exec_one(struct mdio_device *dev, int *argc, char ***argv)
 		mdio_prog_push(&prog, INSN(EMIT, REG(0), 0, 0));
 	}
 
+	printf("%s:3\n", __func__);
+
 	err = mdio_xfer_timeout(dev->bus, &prog, mdio_common_dump_cb, &range, 10000);
 	free(prog.insns);
 	if (err) {
@@ -540,6 +552,8 @@ int mdio_common_dump_exec_one(struct mdio_device *dev, int *argc, char ***argv)
 		return 1;
 	}
 
+	printf("%s:4\n", __func__);
+
 	return 0;
 }
 
@@ -613,17 +627,22 @@ int mdio_xfer_timeout(const char *bus, struct mdio_prog *prog,
 	struct nlmsghdr *nlh;
 	int err;
 
+	printf("%s:1\n", __func__);
 	nlh = msg_init(MDIO_GENL_XFER, NLM_F_REQUEST | NLM_F_ACK);
 	if (!nlh)
 		return -ENOMEM;
 
+	printf("%s:2\n", __func__);
 	mnl_attr_put_strz(nlh, MDIO_NLA_BUS_ID, bus);
 	mnl_attr_put(nlh, MDIO_NLA_PROG, prog->len * sizeof(*prog->insns),
 		     prog->insns);
 
+	printf("%s:3\n", __func__);
 	mnl_attr_put_u16(nlh, MDIO_NLA_TIMEOUT, timeout_ms);
 
+	printf("%s:4\n", __func__);
 	err = msg_query(nlh, mdio_xfer_cb, &xfer);
+	printf("%s:5 xfer.err=%d, err=%d\n", __func__, xfer.err, err);
 	return xfer.err ? : err;
 }
 
-- 
2.47.3

