From 667b3e331d5db99bbef3e7123d72d2c0f2a8260d Mon Sep 17 00:00:00 2001
From: Aleksander Jan Bajkowski <olek2@wp.pl>
Date: Sun, 28 Sep 2025 23:31:00 +0200
Subject: [PATCH] test

---
 src/mdio/mdio.c | 26 +++++++++++++++++++++++++-
 1 file changed, 25 insertions(+), 1 deletion(-)

diff --git a/src/mdio/mdio.c b/src/mdio/mdio.c
index 4677db6..5ad7a70 100644
--- a/src/mdio/mdio.c
+++ b/src/mdio/mdio.c
@@ -16,7 +16,7 @@
 
 #include "mdio.h"
 
-static char buf[0x1000] __attribute__ ((aligned (NLMSG_ALIGNTO)));
+static char buf[0x4000] __attribute__ ((aligned (NLMSG_ALIGNTO)));
 static const size_t len = 0x1000;
 static uint16_t mdio_family;
 
@@ -61,9 +61,12 @@ static int msg_recv(struct mnl_socket *nl, mnl_cb_t callback, void *data, int se
 	unsigned int portid;
 	int ret;
 
+	printf("%s:1\n", __func__);
 	portid = mnl_socket_get_portid(nl);
 
+	printf("%s:2\n", __func__);
 	ret = mnl_socket_recvfrom(nl, buf, len);
+	printf("%s:3 ret=%d\n", __func__, ret);
 	while (ret > 0) {
 		ret = mnl_cb_run(buf, ret, seq, portid, callback, data);
 		if (ret <= 0)
@@ -71,6 +74,7 @@ static int msg_recv(struct mnl_socket *nl, mnl_cb_t callback, void *data, int se
 		ret = mnl_socket_recvfrom(nl, buf, len);
 	}
 
+	printf("%s:4 ret=%d\n", __func__, ret);
 	return ret;
 }
 
@@ -80,15 +84,23 @@ static int msg_query(struct nlmsghdr *nlh, mnl_cb_t callback, void *data)
 	struct mnl_socket *nl;
 	int ret;
 
+	printf("%s:1\n", __func__);
+
 	seq = time(NULL);
 	nlh->nlmsg_seq = seq;
 
+	printf("%s:2\n", __func__);
+
 	nl = msg_send(NETLINK_GENERIC, nlh);
 	if (!nl)
 		return -ENOTSUP;
 
+	printf("%s:3\n", __func__);
+
 	ret = msg_recv(nl, callback, data, seq);
 	mnl_socket_close(nl);
+
+	printf("%s:4 ret=%d\n", __func__, ret);
 	return ret;
 }
 
@@ -519,10 +531,12 @@ int mdio_common_dump_exec_one(struct mdio_device *dev, int *argc, char ***argv)
 	uint32_t reg;
 	int err;
 
+	printf("%s:1\n", __func__);
 	err = mdio_device_parse_reg(dev, argc, argv, &range.start, &range.end);
 	if (err)
 		return err;
 
+	printf("%s:2\n", __func__);
 	/* Can't emit a loop, since there's no way to pass the (mdio)
 	 * register in a (mdio-netlink) register - so we unroll it. */
 	for (reg = range.start; reg <= range.end; reg++) {
@@ -533,6 +547,8 @@ int mdio_common_dump_exec_one(struct mdio_device *dev, int *argc, char ***argv)
 		mdio_prog_push(&prog, INSN(EMIT, REG(0), 0, 0));
 	}
 
+	printf("%s:3\n", __func__);
+
 	err = mdio_xfer_timeout(dev->bus, &prog, mdio_common_dump_cb, &range, 10000);
 	free(prog.insns);
 	if (err) {
@@ -540,6 +556,8 @@ int mdio_common_dump_exec_one(struct mdio_device *dev, int *argc, char ***argv)
 		return 1;
 	}
 
+	printf("%s:4\n", __func__);
+
 	return 0;
 }
 
@@ -613,17 +631,23 @@ int mdio_xfer_timeout(const char *bus, struct mdio_prog *prog,
 	struct nlmsghdr *nlh;
 	int err;
 
+	printf("%s:1\n", __func__);
 	nlh = msg_init(MDIO_GENL_XFER, NLM_F_REQUEST | NLM_F_ACK);
 	if (!nlh)
 		return -ENOMEM;
 
+	printf("%s:2\n", __func__);
 	mnl_attr_put_strz(nlh, MDIO_NLA_BUS_ID, bus);
+	printf("%s:3 len=%d, size=%d\n", __func__, prog->len, sizeof(*prog->insns));
 	mnl_attr_put(nlh, MDIO_NLA_PROG, prog->len * sizeof(*prog->insns),
 		     prog->insns);
 
+	printf("%s:4\n", __func__);
 	mnl_attr_put_u16(nlh, MDIO_NLA_TIMEOUT, timeout_ms);
 
+	printf("%s:5\n", __func__);
 	err = msg_query(nlh, mdio_xfer_cb, &xfer);
+	printf("%s:6 xfer.err=%d, err=%d\n", __func__, xfer.err, err);
 	return xfer.err ? : err;
 }
 
-- 
2.47.3

